### DISCUSSION TITLE
Mounting the components, tapping holes

### DISCUSSION THREAD

**Discussion by daveclifford** [DCC-EX/EX-T3-WiFi#15] [Q&A] [answered]
Alex, 

I received the case parts from my fabricator and am ready to start parts assembly.

I see the center points for the screws to hold the components ,and assume I need to clearance drill for the M2, M3, etc. taps. 

1. Any suggestions on stopping the drill movement so the drill will not break through (and destroy the case)  as well as keeping the drill plumb and level? This applys for the tap downward movement as well.
2. What is the orientation of the on/off switch? Facing the user, is on to the left (or right)?  I.E -  Is the on condition to the left? (or right?)  The switch connections on the back are in a W pattern where the center connection is slightly higher that the outside pair.

(I like the colored confetti)

Dave

**Comment by alex-code** [comment]
I just screwed straight into the plastic.
The holes are slightly undersized so a self tapping screw will create its own thread.

The switch orientation is down to preference.
The side that makes contact will turn it off.

How I did it below in that position it will be off.
![image](https://user-images.githubusercontent.com/1797626/207932436-2ec6f9c5-5502-4c26-8732-a1fb5c8520ad.png)

**Reply by daveclifford** [reply]
Hmm! Reverse logic !  Normally a switch making contact turns something on! 

Thanks for the response!

I suggest you suggest using hex flat head self tapping cap screws to mount the components in your aassembly.md.

I bought a box of  M2-M5 hex flat head cap screws  (NOT self-tapping) that have to be tapped. I think self-tapping screws are the way to go.

Dave Clifford

**Reply by alex-code** [reply]
Hi Dave,

I've updated the docs and included a picture.
I did originally think about using threaded inserts but that would've meant making the case larger and more complex to assemble.

The switch is somewhat reversed as it doesn't break the circuit with the battery but turns off the 3.3v regulator on the FireBeetle.
This allows the battery to still be charged via the USB when turned off.

**Comment by daveclifford** [comment, marked as answer]
I saw the updates to the screws in your readme. Good job to help others in
building the throttle. I've discovered that the 2 mmx 5 cap screw is an odd
size - most screw assortments are in even increments (i.e x2, x4, x6, etc.)
I like the picture you added of the screws with the metric ruler!!!)

Any newcomer may be overwhelmed by the code  tools setup and execution. I
coded for 10 years in IBM from 1973-83 (retired in 2013) w/o the benefit of
IDEs. The Arduino IDE is neat - the Visual Code app is more difficult to
navigate. A how-to may help ( push this, then push that, etc.) - I will
write one for you to include for Linux (ubuntu) - including
installation, setup  and use. I am having trouble loading the code - the
build worked fine)  - the FireBeetle is complaining about not being in the
boot mode for code loading. - It might be that I need to push one of the
buttons.

Dave

**Reply by alex-code** [reply]
Arduino IDE is better for beginners but PlatformIO allows me to provide some more automated setup.
There's certain `#define`s and library dependencies that PlatformIO will manage for you where with Arduino IDE you'd need to download it.

A Linux based guide would be useful, I've only used WSL with VS Code for compilation.

Not sure about the uploading issues with it being Ubuntu, maybe a driver issue?
Does a simple blink program upload?

**Comment by daveclifford** [comment]
Thanks, I'll check it out.  good suggestion

Dave

**Comment by daveclifford** [comment]
I decided to add a 1 MM washer to the screw stack to ensure the 6 mm screw
would provide 5mm of length.

Dave

**Comment by daveclifford** [comment]
I printed the assembly.md file on my Brother laser color printer, so I
could follow the assembly on my hobby workbench.

The printed format is different than presented by the  Google Chrome
browser.

All the unordered lists are scrunched into one sentence and the battery
compartment page did not print well..

I then used an add-on app Print Friendly and PDF to create a PDF file.
(attached below) There are now page numbers and all prints as formatted.

(I like the pics of the stripped insulation spaghetti-firebeetle.jpg)

FYI  and Kind Regards,

Dave in America
